src/firefly/js/visualize/ui/CatalogConstraintsPanel.jsx

```diff
@@ -166,77 +199,147 @@ export class CatalogConstraintsPanel extends React.Component {
 }
 
 /**
- * Set true checkboxes column wherever a row should be selected given by column named 'sel'
- * @param anyTableModel
+ * @summary update the column width in the table model
+ * @param {TableModel} anyTableModel
+ * @param {string[]} colName
+ * @param {number} colWidth
  */
-function setRowsChecked(anyTableModel) {
-    const selectInfoCls = SelectInfo.newInstance({rowCount: anyTableModel.totalRows});
-    const idxColSel = getColumnIdx(anyTableModel, 'sel');
-    anyTableModel.tableData.data.forEach((arow, index) => {
-        if (arow[idxColSel] === 'y') {
-            selectInfoCls.setRowSelect(index, true);
+function updateColumnWidth(anyTableModel, colName, colWidth) {
+    if (!isArray(colName)) {
+        colName = [colName];
+    }
+
+    colName.find((cName) => {
+        var idx = getColumnIdx(anyTableModel, cName);
+
+        if (idx >= 0) {
+            if (colWidth < 0) {
+                var w = anyTableModel.tableData.data.reduce((prev, d) => {
+                    if (d[idx].length > prev)  {
+                        prev = d[idx].length;
+                    }
+                    return prev;
+                }, colWidth);
+                colWidth = w;
+            }
+            anyTableModel.tableData.columns[idx].width = colWidth;
         }
+        return (idx >= 0);
     });
+}
+/**
+ * Set true checkboxes column wherever a row should be selected based on the stored status or the 'sel' value
+ * the 'sel' value for each column is set as the following condition:
+ * - table model is acquired the first time: add 'sel' column with all 'y' if 'sel' column doesn't exit
+ * - table model status is kept already: the 'sel' column value may be updated by column selection operation,
+ *                                       set the value of SelectInfo as the column 'sel' value
+ * @param {TableModel} anyTableModel
+ * @param {string} gkey group key
+ */
+function setRowsChecked(anyTableModel, gkey) {
+    var {selcols = '', errorConstraints} = get(FieldGroupUtils.getGroupFields(gkey), 'tableconstraints.value', {});
+    var filterAry = selcols ? selcols.split(',') : [];
+    const selectInfoCls = SelectInfo.newInstance({rowCount: anyTableModel.totalRows});
+    var idxColSel = getColumnIdx(anyTableModel, 'sel');
+
+    if (idxColSel < 0) {     // no 'sel' column exist
+        idxColSel = addSelColumn(anyTableModel, filterAry, errorConstraints);
+    }
+    if (filterAry.length > 0 || !isEmpty(errorConstraints)) {  // get kept status (some selected or none selected)
+        anyTableModel.tableData.data.forEach((arow, index) => {
+            if (filterAry.includes(arow[0])) {
+                selectInfoCls.setRowSelect(index, true);
+            }
+        });
+    } else { // the table model may be freshly acquired or column selection status is updated as all columns are deselected
+        anyTableModel.tableData.data.forEach((arow, index) => {
+            if (arow[idxColSel] === 'y') {
```

Comment:
sometime it is 'n' in case the column is deselected or initially defined as deselected.